nipoppy_new/nipoppy/base.py

```diff
@@ -0,0 +1,71 @@
+"""Base class."""
```

Comment:
I would not worry about this because given that each is its own subpackage, they should live in their own namespace.

I'd be surprised if we needed to import different `base` into the same module, unless we start doing some funky frankensteiny multiple inheritance stuff. And even then we could alias one of the `base`:

```python
from nipoppy import base
from nipoppy.workflows import base as wk_base
```